src/components/ShoppingCart/ShoppingCartContainer.tsx

```diff
@@ -37,6 +42,16 @@ const ShoppingCartContainer = ({ shoppingCart }: userCartList) => {
   const [cartItemIdToDelete, setCartItemIdToDelete] = useState("");
   const anyExpired = shoppingCart.some((item) => item.isExpired);
 
+  const [checkedItems, setCheckedItems] = useState(shoppingCart.map(() => false));
+  const allChecked = checkedItems.every(Boolean);
+  const isIndeterminate = checkedItems.some(Boolean) && !allChecked;
+
+  const handleCreateOrder = () => {
+    const orders = shoppingCart.filter((e, index) => checkedItems[index] === true);
```

Comment:
it can be replaced with 
_const orders = shoppingCart.filter((e, index) => checkedItems[index]);_